pkg/app/PipelineStatusTimelineService.go

```diff
@@ -44,6 +56,65 @@ type PipelineStatusTimelineDto struct {
 	Status                       pipelineConfig.TimelineStatus `json:"status"`
 	StatusDetail                 string                        `json:"statusDetail"`
 	StatusTime                   time.Time                     `json:"statusTime"`
+	ResourceDetails              []*SyncStageResourceDetailDto `json:"resourceDetails,omitempty"`
+}
+
+func (impl *PipelineStatusTimelineServiceImpl) SaveTimeline(timeline *pipelineConfig.PipelineStatusTimeline, tx *pg.Tx) error {
+	//get unableToFetch or timedOut timeline
+	redundantTimelines, err := impl.pipelineStatusTimelineRepository.FetchTimelineByWfrIdAndStatuses(timeline.CdWorkflowRunnerId, []pipelineConfig.TimelineStatus{pipelineConfig.TIMELINE_STATUS_UNABLE_TO_FETCH_STATUS, pipelineConfig.TIMELINE_STATUS_FETCH_TIMED_OUT})
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("error in getting unableToFetch/timedOut timelines", "err", err, "cdWfrId", timeline.CdWorkflowRunnerId)
+		return err
+	}
+	if len(redundantTimelines) > 1 {
+		return fmt.Errorf("multiple unableToFetch/timedOut timelines found")
+	} else {
+		if len(redundantTimelines) == 1 && redundantTimelines[0].Id > 0 {
```

Comment:
instead of nested else use `if else` here

Reply:
done